File: p25/p25.py
#!/usr/bin/python3
import math

phi = (1 + pow(5, 0.5)) / 2
c = math.log10(5) / 2
logphi = math.log10(phi)
n = 1
while True:
    if n * logphi - c >= 999:
        break
    n = n + 1

print("Answer %s " % n)

File: p25/p25.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname p25
% vim:syn=erlang

-mode(compile).


main(_) ->
    io:format("Answer: ~p ~n", [test25()]).

test25()     -> fib(1,2,2).
fib(A,B,N)   ->
    case length(integer_to_list(A)) of
        1000 -> N;
        _    -> fib(B, A+B,N+1)
    end.

